src/gedi_subset/gedi_utils.py

```diff
@@ -61,6 +62,46 @@ def gdf_read_parquet(path: Union[str, os.PathLike[str]]) -> IOResultE[gpd.GeoDat
     return impure_safe(gpd.read_parquet)(path)
 
 
+def gdf_reformat(path: Union[str, os.PathLike[str]], suffix: str) -> None:
+    """
+    Convert a GeoDataFrame written in one format to another format.
+
+    Do nothing if the `suffix` property of the path is the same as the specified
+    suffix.  Otherwise, read the GeoDataFrame at the specified path and write it
+    back out next to the original file, but with the specified suffix, and then
+    remove the original file.
+
+    Parameters
+    ----------
+    path:
+        Path to an existing GeoDataFrame written to the filesystem.
+    suffix:
+        Desired format, expressed as a destination file suffix, including a
+        leading dot (`.`), just like the `suffix` property of a `Path` includes.
+        Supported values: ".parquet", ".feather", or any suffix supported by
+        `GeoDataFrame.to_file`, which will use the suffix to guess the
+        appropriate driver.
+    """
+    src_path = path if isinstance(path, Path) else Path(path)
+    dst_path = src_path.with_suffix(suffix)
+
+    # No need to waste time when the file is already in the desired format
+    if src_path.suffix == suffix:
+        return
+
+    gdf: gpd.GeoDataFrame = gpd.read_file(path)
```

Comment:
Is this the best approach? I suppose it makes more sense then shelling to ogr2ogr but I wonder if the memory might be an issue since fgb, and maybe in the future parquet need to calculate spatial indexes? @kylebarron have a better idea?

Reply:
Good question. Memory requirements for reading the entire subset result into memory did cross my mind. However, rather than shelling out, my thought was to call ogr2ogr from the "run" shell script _after_ the Python process completed.

Either way, would ogr2ogr require significantly less RAM to convert a file from gpkg to another format than reading the file in Python and then writing it back out to another format would require? I suspect ogr2ogr would indeed require less RAM, but would it be _significantly_ less, such that ogr2ogr would avoid an OOM error that the Python code might otherwise encounter?

Reply:
We decided this isn't a blocker, since the GPKG results will still work it's a non breaking change if something goes wrong.